### DISCUSSION TITLE
React native compatibility

### DISCUSSION THREAD

**Discussion by devnomic** [Omnistac/zedux#74] [General]
@bowheart ,
I want to use this lib for my rn project, as i'm pretty sure this lib can be used for react native, but when i tried it shows error when run it.

it seems cannot resolve the zedux core somehow.
I hope there is a way to solve this, this lib is too good, i can't go back to jotai or other redux state management now. 😅  

````
Metro has encountered an error: While trying to resolve module `@zedux/core` from file 
`/Users/erick/Documents/RNProjects/ExampleApp/node_modules/@zedux/atoms/dist/cjs/index.js`, the package 
`/Users/erick/Documents/RNProjects/ExampleApp/node_modules/@zedux/core/package.json` was successfully found. 
However, this package itself specifies a `main` module field that could not be resolved 
(`/Users/erick/Documents/RNProjects/ExampleApp/node_modules/@zedux/core/dist/zedux.umd.js`. 
Indeed, none of these files exist:

  * /Users/erick/Documents/RNProjects/ExampleApp/node_modules/@zedux/core/dist/zedux.umd.js(.native|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx)
  * /Users/erick/Documents/RNProjects/ExampleApp/node_modules/@zedux/core/dist/zedux.umd.js/index(.native|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx): /Users/erick/Documents/RNProjects/ExampleApp/node_modules/metro/src/node-haste/DependencyGraph.js (289:17)

  287 |         }
  288 |         if (error instanceof InvalidPackageError) {
> 289 |           throw new PackageResolutionError({
      |                 ^
  290 |             packageError: error,
  291 |             originModulePath: from,
  292 |             targetModuleName: to,

RCTFatal
__28-[RCTCxxBridge handleError:]_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_drain
_dispatch_main_queue_callback_4CF
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
-[UIApplication _run]
UIApplicationMain
main
start_sim
0x0


````

**Comment by bowheart** [comment]
Hey! Thanks for reporting.

Indeed, looks like the `main` field of the core package's package.json is pointing to the old location. We probably haven't run into this because the `exports.default` and `exports.require` fields both point to the correct file. I'd expect all bundlers to be able to resolve those by now. Very odd that Metro doesn't.

I will publish a `v1.1.0-rc.0` prerelease with that fix. You should be able to install it with `npm i @zedux/react@next`.

Other than that, to my knowledge, RN _should_ work. I haven't tried it yet, so definitely feel free to add to this discussion if you find anything else.

**Reply by bowheart** [reply]
@devnomic `v1.1.0-rc.0` is published. If you can install it (`npm i @zedux/react@next`) and verify this is fixed, I don't have anything else preventing `v1.1.0` from being released. If there are other problems, I can publish more release candidate versions until we get them resolved.

**Reply by devnomic** [reply]
I tried the `v1.1.0-rc.0`, works now! Thank you!

**Reply by bowheart** [reply]
Awesome! Will plan on releasing v1.1.0 very soon then. Today if I get to it.

**Reply by bowheart** [reply]
@devnomic v1.1.0 is released btw. I was trying to maybe slip in some more features, but it was taking too long so I finally just cut the release. More stuff coming in v1.2.0
